src/Entity/PurchasedPlan.php

```diff
@@ -271,14 +271,14 @@ public function getTermsAndConditions(): bool {
    */
   public function getStatus(): string {
```

Comment:
Can we have this covered with unit tests also and cover all possible scenarios? Like when the site's timezone is different than Apigee's timezone, etc.

Reply:
Yes definitely. We have road-map for adding various test cases. I will create this as a separate issue so it can be addressed with other test coverage tasks. Thanks @mxr576.

Reply:
I do not see why it should not be added now as there is a known problem with the current implementation, covering the fix with tests before this CR is merged is (IMO) the best way to ensure this bug never occurs again, but ofc, it is up to the team how tasks are being scheduled. :) I am already glad that this issue got a fix.

Reply:
@mxr576 very good point, we will get back to adding/updating tests with a PR to get it accepted. FYI @raakesh-blokhra @giteshk